dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -2942,25 +2942,29 @@ Spdp::SpdpTransport::join_multicast_group(const DCPS::NetworkInterface& nic,
                  all_interfaces ? "all interfaces" : nic.name().c_str()));
     }
 
-    if (0 == multicast_socket_.join(multicast_address_, 1, all_interfaces ? 0 : ACE_TEXT_CHAR_TO_TCHAR(nic.name().c_str()))) {
-      joined_interfaces_.insert(nic.name());
+    static ACE_Thread_Mutex ipv4_static_lock;
+    {
```

Comment:
Does this require an "artificial" scope here?  (It's hard to tell from the diffs on GitHub, if the code would be equivalent without it then leave it out.)